odoo/modules/loading.py

```diff
@@ -191,20 +191,22 @@ def load_module_graph(
                 registry._setup_models__(env.cr)
                 getattr(py_module, pre_init)(env)
 
-        model_names = registry.load(env.cr, package)
+        model_names_iter = registry.load(package)
```

Comment:
I don't think this improves performance, why do you want to change this?
For me, it makes the code harder to read.

Reply:
If the call to `descendants()` is sometimes unnecessary, then remove it from method `load()` and call it explicitly where needed.  `load()` would return the names of the models directly impacted by the loading.

Reply:
Or, as an alternative, return a callable that returns all the impacted model names.  And you simply call it if you need it.